sim/src/Manager.h

```diff
@@ -0,0 +1,46 @@
+#ifndef BENTOBOX_MANAGER_H
+#define BENTOBOX_MANAGER_H
+
+#include <memory>
+#include <ics/index.h>
+#include <ics/component.h>
+#include <ics/compVec.h>
+#include <util/typeMap.h>
+
+namespace {
+    template<ics::Component C>
+    using CompContainer = ics::CompVec<C>;
+}
+
+class Manager {
+private:
+    std::unique_ptr<util::TypeMap> indexes;
+    std::unique_ptr<util::TypeMap> components;
+public:
+    template<ics::Component C>
+    void addComponent(C c) {
+        if (components->has<CompContainer<C>>()) {
+            // The container already exists
+            // TODO: test that this actually updates the container value
+            auto container = components->at<CompContainer<C>>();
+            container.add(c);
+        } else {
+            // The container does not exist
+            auto container = CompContainer<C>();
+            container.add(c);
+            components->insert(container);
+        }
+    }
+
+    template<ics::Component C, ics::Index I>
+    void addIndex(I i) {
```

Comment:
Assuming refers to index in ECS. If so it would be nice to stick to Entity naming convention.

Reply:
I assume you mean
> Assuming this refers to **entity** in ECS ...
An index is not the same as an entity. An entity in ECS is a single unsigned int which multiple components are associated with. However, an index is a more general idea. An index is the association of any component to anything. Similar to an index in databases, it stores some key which makes retrieval faster.

For example, I can have an `EntityIndex` which works the same as entities and I can have a `TagIndex` which maps a single tag string to components or entities.